ztools/x509/x509.go

```diff
@@ -1135,51 +1202,87 @@ func parseCertificate(in *certificate) (*Certificate, error) {
 				for _, subtree := range constraints.Permitted {
 					switch subtree.Value.Tag {
```

Comment:
It is the same for all generalNames, so it also applies to SAN and other extensions that use generalName.